diagnostics/collect-wsl-logs.ps1

```diff
@@ -130,6 +130,72 @@ if ($Dump)
     }
 }
 
+# Collect networking state relevant for WSL
+# Using a try/catch for commands below, as some of them do not exist on all OS versions
+
+try
+{
+    Get-NetAdapter -includeHidden | select Name,ifIndex,NetLuid,InterfaceGuid,Status,MacAddress,MtuSize,InterfaceType,Hidden,HardwareInterface,ConnectorPresent,MediaType,PhysicalMediaType | Out-File -FilePath "$folder/Get-NetAdapter.log" -Append
+}
+catch {}
+
+try
+{
+    Get-NetFirewallHyperVVMCreator | Out-File -FilePath "$folder/Get-NetFirewallHyperVVMCreator.log" -Append
+}
+catch {}
+
+try
+{
+    Get-NetFirewallHyperVVMSetting -PolicyStore ActiveStore | Out-File -FilePath "$folder/Get-NetFirewallHyperVVMSetting_ActiveStore.log" -Append
+}
+catch {}
+
+try
+{
+    Get-NetFirewallHyperVProfile -PolicyStore ActiveStore | Out-File -FilePath "$folder/Get-NetFirewallHyperVProfile_ActiveStore.log" -Append
+}
+catch {}
+
+try
+{
+    Get-NetFirewallHyperVPort | Out-File -FilePath "$folder/Get-NetFirewallHyperVPort.log" -Append
+}
+catch {}
+
+try
+{
+    & hnsdiag.exe list all 2>&1 > $folder/hnsdiag_list_all.log
+}
+catch {}
+
+try
+{
+    & hnsdiag.exe list endpoints -df 2>&1 > $folder/hnsdiag_list_endpoints.log
+}
+catch {}
+
+try
+{
+    foreach ($port in Get-NetFirewallHyperVPort)
+    {
+        $vfpLogFile = "$folder/vfp-port-" + $port.PortName + "-get-port-state.log"
+        $cmd = "vfpctrl.exe /port " + $port.PortName + " /get-port-state > " + $vfpLogFile
```

Comment:
Could this also use the &<command> 2>&1 > $folder/<log> structure ?

Reply:
I wasn't able to get the right syntax here using operator "&" - since the command string needs to be dynamically created, I tried doing just & $dynamic_cmd_variable, but this did not seem to work

Reply:
Should be resolved with latest fix

Reply:
Oh, after taking a look at what this was previously, did you want this to be called using "& vfpctrl.exe ..." rather than use the $cmd | cmd | Out-Null and just add "2>&1 >" to the commands? I can give this a try

Reply:
okay, I think i found right escape characters, so should be resolved with latest push